<?xml version="1.0"?>
<launch>
    <node pkg="tl_detector" type="tl_detector.py" name="tl_detector" output="screen" cwd="node">

    <!--Traffic Light Detection Node Params -->

    <!--Traffic Light Detection Model Path -->
    <param name="traffic_light_detector_model" type="str" value="$(find tl_detector)/models/mobilenet_ssd/frozen_inference_graph.pb" />

    <!--Traffic Light Training Data Path -->
    <param name="traffic_light_training_data_directory" value="$(find tl_detector)/training_data" />

    <!--Use the sim_detector false means use the real detector -->
    <param name="use_sim_detector" value="False" />

    <!--Save images when camera is on -->
    <param name="save_data" type="bool" value="False" />

    <!--Run detector debug mode -->
    <param name="run_debug" type="bool" value="False" />

    <!--Temporal filter thr.  How many times must see traffic light state before say that is state externally-->
    <param name="state_count_threshold" type="int" value="2" />

    <!--How confident we need to be in a detection before counting it-->
    <param name="confidence_thr" type="double" value="0.20" />



    </node>
    <node pkg="tl_detector" type="light_publisher.py" name="light_publisher" output="screen" cwd="node"/>
</launch>
